src/libraries/System.Private.Xml/src/System/Xml/Core/XmlReader.cs

```diff
@@ -1153,46 +1128,42 @@ public virtual string ReadInnerXml()
             {
                 return string.Empty;
             }
+
             if (NodeType != XmlNodeType.Attribute && NodeType != XmlNodeType.Element)
             {
                 Read();
+
                 return string.Empty;
             }
 
             StringWriter sw = new(CultureInfo.InvariantCulture);
-            XmlWriter xtw = CreateWriterForInnerOuterXml(sw);
+            using XmlTextWriter xtw = CreateWriterForInnerOuterXml(sw);
 
-            try
+            if (NodeType == XmlNodeType.Attribute)
```

Comment:
the removal of try-finally may be observable, it's not strictly a refactoring.

Reply:
@Wraith2 
https://github.com/dotnet/runtime/blob/4ea6120cb223adf6ff693b9f6ebcda41c613e3ec/src/libraries/System.Private.Xml/src/System/Xml/Core/XmlReader.cs#L1140

Reply:
Oh, the hidden `using` scope syntax. I'm not a fan of that I like to see the scopes nice and explicitly. We'll see what others think.

Reply:
@Wraith2 
the area is clearly visible here, but let's see what others say

Reply:
looks fine with me as is in the PR already

Reply:
actually backing off with that - if XmlTextWriter is not closed before calling ToString you might end up with the node not closed (i.e. if you start the element but not finish it the Dispose/Close will close it but now it won't) - I'm ok if you add test cases with that and ensure this actually happens. Scoped `using` sounds better to me here to be safe